libs/beacon/tests/dkg/beacon_manager_tests.cpp

```diff
@@ -0,0 +1,378 @@
+  {
+    manager->Reset(cabinet, threshold);
+  }
+
+  // Check reset for one manager
+  for (uint32_t index = 0; index < cabinet_size; ++index)
+  {
+    std::shared_ptr<BeaconManager> manager = beacon_managers[index];
+    EXPECT_EQ(manager->cabinet_index(),
+              std::distance(cabinet.begin(), cabinet.find(member_ptrs[index]->identity())));
+    for (auto &mem : member_ptrs)
+    {
+      EXPECT_EQ(manager->cabinet_index(mem->identity().identifier()),
+                std::distance(cabinet.begin(), cabinet.find(mem->identity())));
+    }
+    EXPECT_TRUE(manager->qual().empty());
+    EXPECT_TRUE(manager->group_public_key() == zero.getStr());
+  }
+
+  for (auto &manager : beacon_managers)
+  {
+    manager->GenerateCoefficients();
+  }
+
+  // Checks
+  for (uint32_t index = 0; index < cabinet_size; ++index)
+  {
+    std::shared_ptr<BeaconManager> manager      = beacon_managers[index];
+    std::vector<std::string>       coefficients = manager->GetCoefficients();
+    for (uint32_t elem = 0; elem < threshold; ++elem)
+    {
+      // Coefficients generated should be non-zero
+      EXPECT_NE(coefficients[elem], zero.getStr());
+      for (uint32_t index1 = 0; index1 < cabinet_size; ++index1)
+      {
+        if (index1 != index)
+        {
+          std::shared_ptr<BeaconManager> another_manager = beacon_managers[index1];
+          std::vector<std::string>       coefficients1   = another_manager->GetCoefficients();
+          // Coefficients should be different
+          EXPECT_NE(coefficients, coefficients1);
+          // Shares generated ourself are non-zero
+          EXPECT_NE(manager->GetOwnShares(member_ptrs[index1]->identity().identifier()).first,
+                    zero.getStr());
+          EXPECT_NE(manager->GetOwnShares(member_ptrs[index1]->identity().identifier()).second,
+                    zero.getStr());
+          // Shares received from others should be zero
+          EXPECT_EQ(manager->GetReceivedShares(member_ptrs[index1]->identity().identifier()).first,
+                    zero.getStr());
+          EXPECT_EQ(manager->GetReceivedShares(member_ptrs[index1]->identity().identifier()).second,
+                    zero.getStr());
+        }
+      }
+    }
+  }
+
+  // Name identifiers
+  MuddleAddress my_address = addresses[0];
+  MuddleAddress honest     = addresses[1];
+  MuddleAddress malicious  = addresses[2];
+
+  // Add shares and coefficients passing verification from someone and check that they are entered
+  // in correctly
+  beacon_managers[0]->AddShares(honest, beacon_managers[1]->GetOwnShares(my_address));
+  beacon_managers[0]->AddCoefficients(honest, beacon_managers[1]->GetCoefficients());
+  EXPECT_EQ(beacon_managers[0]->GetReceivedShares(honest),
+            beacon_managers[1]->GetOwnShares(my_address));
+
+  // Add shares and coefficients failing verification from malicious party
+  PrivateKey s_i;
+  PrivateKey sprime_i;
+  s_i.setStr(beacon_managers[2]->GetOwnShares(my_address).first);
```

Comment:
this line is a bit confusing. Why is getownshares taking an address?
I would probably split this into two lines to make it a bit clearer what the .first refers to

Reply:
GetOwnShares takes an address because it returns the pair of shares computed for and sent to a particular cabinet member. I will split into two lines.